sdk/iot/hub/src/az_iot_hub_client_methods.c

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include "az_iot_hub_client.h"
+#include <az_precondition_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+#include <_az_cfg.h>
+
+static const az_span methods_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("$iothub/methods/");
```

Comment:
See below: some reordering and splicing here will simplify the code.

Reply:
Do you mean this is related to https://github.com/Azure/azure-sdk-for-c/pull/559#discussion_r406509514 ?
I will save that as an improvement comment, which I want to compare on the compiler before taking.